sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -172,6 +172,8 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
     // optimizer rules that are triggered when there is a filter
     // (e.g. InferFiltersFromConstraints). If we run this batch earlier, the query becomes just
     // LocalRelation and does not trigger many rules.
```

Comment:
The comment should be before the batch `LocalRelation early`.

Reply:
@gengliangwang, fixed this.